File: 049.LeetCode@56_Merge_Intervals/049.LeetCode@56_Merge_Intervals.py
class Solution:
    def merge(self, intervals: list[list[int]]) -> list[list[int]]:
        intervals.sort(key=lambda x: (x[0], x[1]))
        start = intervals[0][0]
        end = intervals[0][1]
        ans=[]
        for i in range(1,len(intervals)):
            if intervals[i][0] > end:
                ans.append([start, end])
                start = intervals[i][0]
                end = intervals[i][1]
            else:
                end =max(intervals[i][1],end)
        ans.append([start, end])
        return ans

File: 049.LeetCode@56_Merge_Intervals/049.LeetCode@56_Merge_Intervals.java
class Solution {
    public static int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                int compare = Integer.compare(o1[0], o2[0]);
                if (compare != 0) {
                    return compare;
                } else {
                    return Integer.compare(o2[1], o1[1]);
                }
            }
        });
        List<int[]> list = new ArrayList<>();
        int start = intervals[0][0];
        int end = intervals[0][1];
        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] > end) {
                list.add(new int[]{start, end});
                start = intervals[i][0];
                end = intervals[i][1];
            } else {
                end = Math.max(end, intervals[i][1]);
            }
        }
        list.add(new int[]{start, end});
        return list.toArray(new int[list.size()][]);
    }
}